### PR TITLE
Rebuild with OpenSSL 3 [skip ci]

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Rebuild with OpenSSL 3. 2.1.12 isn't officially compatible with OpenSSL 3 and requires a small patch that I took from Arch Linux (which they themselves took from some PRs in libevent).

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,12 +30,10 @@ requirements:
...
     - m2-patch    # [win]
     - {{ compiler('c') }}
-    - openssl
   host:
-    - openssl
+    - openssl {{ openssl }}
   run:
-    # Will be taken care of by run_exports
-    # - openssl 1.0.*
+    - openssl 3.*
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:36]: maybe pin_compatibile would be better and more generic ?

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Review by varlackc — COMMENTED** [review]
We might want to remove the abs.yaml file before merging to defaults

**Comment by danpetry** [comment]
@varlackc do you have any remaining changes required to be addressed, or are you able to give this an approval if not?

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**